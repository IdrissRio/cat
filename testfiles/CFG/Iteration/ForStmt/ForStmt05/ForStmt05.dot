digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[22]"; "Opt[25]"; "List[26]"; "List[27]"; "InstanceInitialization[6]"; "InstanceInitialization[14]"}
{ rank = same; "List[23]"; "InstanceInitializer[28]"; "Entry[7]"; "Exit[8]"; "Entry[15]"; "Exit[16]"}
{ rank = same; "Modifier[24]"; "Block[29]"}
{ rank = same; "List[30]"}
{ rank = same; "ForStmt[31]"}
{ rank = same; "List[32]"; "Opt[46]"; "List[50]"; "Block[57]"}
{ rank = same; "VarDeclStmt[33]"; "EQExpr[47]"; "ExprStmt[51]"; "ExprStmt[54]"; "List[58]"}
{ rank = same; "Modifiers[34]"; "PrimitiveTypeAccess[36]"; "List[37]"; "VarAccess[48]"; "IntegerLiteral[49]"; "PostIncExpr[52]"; "PostIncExpr[55]"; "ExprStmt[59]"}
{ rank = same; "List[35]"; "VariableDeclarator[38]"; "VariableDeclarator[42]"; "VarAccess[53]"; "VarAccess[56]"; "Dot[60]"}
{ rank = same; "List[39]"; "Opt[40]"; "List[43]"; "Opt[44]"; "Dot[61]"; "MethodAccess[64]"}
{ rank = same; "IntegerLiteral[41]"; "IntegerLiteral[45]"; "TypeAccess[62]"; "VarAccess[63]"; "List[66]"; "List[68]"; "List[69]"; "UncheckedExceptions[65]"}
{ rank = same; "IntegerLiteral[67]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < ForStmt05 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[22]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[24]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[25]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[27]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitializer[28]" [label="InstanceInitializer\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[29]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[30]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ForStmt[31]" [label="ForStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[32]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[33]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[34]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[35]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[36]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[38]" [label="VariableDeclarator\n < b >", fillcolor=white   style=filled  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[40]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[41]" [label="IntegerLiteral\n < 2 >", fillcolor=white   style=filled  ]
"VariableDeclarator[42]" [label="VariableDeclarator\n < c >", fillcolor=white   style=filled  ]
"List[43]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[44]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[45]" [label="IntegerLiteral\n < 1 >", fillcolor=white   style=filled  ]
"Opt[46]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"EQExpr[47]" [label="EQExpr\n ", fillcolor=white   style=filled  ]
"VarAccess[48]" [label="VarAccess\n < b >", fillcolor=white   style=filled  ]
"IntegerLiteral[49]" [label="IntegerLiteral\n < 2 >", fillcolor=white   style=filled  ]
"List[50]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[51]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PostIncExpr[52]" [label="PostIncExpr\n < b >", fillcolor=white   style=filled  ]
"VarAccess[53]" [label="VarAccess\n < b >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[54]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PostIncExpr[55]" [label="PostIncExpr\n < c >", fillcolor=white   style=filled  ]
"VarAccess[56]" [label="VarAccess\n < c >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[57]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[58]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[59]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[60]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[61]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[62]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[63]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[64]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[66]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[67]" [label="IntegerLiteral\n < 2 >", fillcolor=white   style=filled  ]
"List[68]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[69]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[65]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"InstanceInitialization[6]" [label="InstanceInitialization\n < Instance >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[7]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[8]" [label="Exit\n ", fillcolor=white   style=filled  ]
"InstanceInitialization[14]" [label="InstanceInitialization\n < Static >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Exit[16]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[23]" -> "Modifier[24]" [style=dashed, color=gray]
"Modifiers[22]" -> "List[23]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[22]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[25]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[26]" [style=dashed, color=gray]
"Modifiers[34]" -> "List[35]" [style=dashed, color=gray]
"VarDeclStmt[33]" -> "Modifiers[34]" [style=dashed, color=gray]
"VarDeclStmt[33]" -> "PrimitiveTypeAccess[36]" [style=dashed, color=gray]
"VariableDeclarator[38]" -> "List[39]" [style=dashed, color=gray]
"Opt[40]" -> "IntegerLiteral[41]" [style=dashed, color=gray]
"VariableDeclarator[38]" -> "Opt[40]" [style=dashed, color=gray]
"List[37]" -> "VariableDeclarator[38]" [style=dashed, color=gray]
"VariableDeclarator[42]" -> "List[43]" [style=dashed, color=gray]
"Opt[44]" -> "IntegerLiteral[45]" [style=dashed, color=gray]
"VariableDeclarator[42]" -> "Opt[44]" [style=dashed, color=gray]
"List[37]" -> "VariableDeclarator[42]" [style=dashed, color=gray]
"VarDeclStmt[33]" -> "List[37]" [style=dashed, color=gray]
"List[32]" -> "VarDeclStmt[33]" [style=dashed, color=gray]
"ForStmt[31]" -> "List[32]" [style=dashed, color=gray]
"EQExpr[47]" -> "VarAccess[48]" [style=dashed, color=gray]
"EQExpr[47]" -> "IntegerLiteral[49]" [style=dashed, color=gray]
"Opt[46]" -> "EQExpr[47]" [style=dashed, color=gray]
"ForStmt[31]" -> "Opt[46]" [style=dashed, color=gray]
"PostIncExpr[52]" -> "VarAccess[53]" [style=dashed, color=gray]
"ExprStmt[51]" -> "PostIncExpr[52]" [style=dashed, color=gray]
"List[50]" -> "ExprStmt[51]" [style=dashed, color=gray]
"PostIncExpr[55]" -> "VarAccess[56]" [style=dashed, color=gray]
"ExprStmt[54]" -> "PostIncExpr[55]" [style=dashed, color=gray]
"List[50]" -> "ExprStmt[54]" [style=dashed, color=gray]
"ForStmt[31]" -> "List[50]" [style=dashed, color=gray]
"Dot[61]" -> "TypeAccess[62]" [style=dashed, color=gray]
"Dot[61]" -> "VarAccess[63]" [style=dashed, color=gray]
"Dot[60]" -> "Dot[61]" [style=dashed, color=gray]
"List[66]" -> "IntegerLiteral[67]" [style=dashed, color=gray]
"MethodAccess[64]" -> "List[66]" [style=dashed, color=gray]
"MethodAccess[64]" -> "List[68]" [style=dashed, color=gray]
"MethodAccess[64]" -> "List[69]" [style=dashed, color=gray]
"MethodAccess[64]" -> "UncheckedExceptions[65]" [style=dashed, color=gray]
"Dot[60]" -> "MethodAccess[64]" [style=dashed, color=gray]
"ExprStmt[59]" -> "Dot[60]" [style=dashed, color=gray]
"List[58]" -> "ExprStmt[59]" [style=dashed, color=gray]
"Block[57]" -> "List[58]" [style=dashed, color=gray]
"ForStmt[31]" -> "Block[57]" [style=dashed, color=gray]
"List[30]" -> "ForStmt[31]" [style=dashed, color=gray]
"Block[29]" -> "List[30]" [style=dashed, color=gray]
"InstanceInitializer[28]" -> "Block[29]" [style=dashed, color=gray]
"List[27]" -> "InstanceInitializer[28]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[27]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Entry[7]" [style=dashed, color=gray]
"InstanceInitialization[6]" -> "Exit[8]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Entry[15]" [style=dashed, color=gray]
"InstanceInitialization[14]" -> "Exit[16]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "InstanceInitialization[14]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[16]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "Exit[16]" [color=blue, constraint=false, xlabel=" "] 
"PostIncExpr[55]" -> "VarAccess[48]" [color=blue, constraint=false, xlabel=" "] 
"PostIncExpr[55]" -> "PostIncExpr[52]" [color=red, constraint=false]
"PostIncExpr[52]" -> "PostIncExpr[55]" [color=blue, constraint=false, xlabel=" "] 
"PostIncExpr[52]" -> "MethodAccess[64]" [color=red, constraint=false]
"MethodAccess[64]" -> "PostIncExpr[52]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[64]" -> "IntegerLiteral[67]" [color=red, constraint=false]
"IntegerLiteral[67]" -> "MethodAccess[64]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[67]" -> "VarAccess[63]" [color=red, constraint=false]
"VarAccess[63]" -> "IntegerLiteral[67]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[63]" -> "TypeAccess[62]" [color=red, constraint=false]
"TypeAccess[62]" -> "VarAccess[63]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[62]" -> "EQExpr[47]" [color=red, constraint=false]
"Exit[8]" -> "EQExpr[47]" [color=red, constraint=false]
"EQExpr[47]" -> "TypeAccess[62]" [color=blue, constraint=false, xlabel="T "] 
"EQExpr[47]" -> "Exit[8]" [color=blue, constraint=false, xlabel="F "] 
"EQExpr[47]" -> "IntegerLiteral[49]" [color=red, constraint=false]
"IntegerLiteral[49]" -> "EQExpr[47]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[49]" -> "VarAccess[48]" [color=red, constraint=false]
"VarAccess[48]" -> "IntegerLiteral[49]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[48]" -> "VariableDeclarator[42]" [color=red, constraint=false]
"VarAccess[48]" -> "PostIncExpr[55]" [color=red, constraint=false]
"VariableDeclarator[42]" -> "VarAccess[48]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[42]" -> "IntegerLiteral[45]" [color=red, constraint=false]
"IntegerLiteral[45]" -> "VariableDeclarator[42]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[45]" -> "VariableDeclarator[38]" [color=red, constraint=false]
"VariableDeclarator[38]" -> "IntegerLiteral[45]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[38]" -> "IntegerLiteral[41]" [color=red, constraint=false]
"IntegerLiteral[41]" -> "VariableDeclarator[38]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[41]" -> "Entry[7]" [color=red, constraint=false]
"Entry[7]" -> "IntegerLiteral[41]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"InstanceInitialization[14]" -> "InstanceInitialization[6]" -> "Modifiers[22]" -> "Opt[25]" -> "List[26]" -> "List[27]" [style=invis]
"List[32]" -> "Opt[46]" -> "List[50]" -> "Block[57]" [style=invis]
"Modifiers[34]" -> "PrimitiveTypeAccess[36]" -> "List[37]" [style=invis]
"VariableDeclarator[38]" -> "VariableDeclarator[42]" [style=invis]
"List[39]" -> "Opt[40]" [style=invis]
"List[43]" -> "Opt[44]" [style=invis]
"VarAccess[48]" -> "IntegerLiteral[49]" [style=invis]
"ExprStmt[51]" -> "ExprStmt[54]" [style=invis]
"Dot[61]" -> "MethodAccess[64]" [style=invis]
"TypeAccess[62]" -> "VarAccess[63]" [style=invis]
"Entry[7]" -> "Exit[8]" [style=invis]
"Entry[15]" -> "Exit[16]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
60", fillcolor=1]
Edges [label="Total Nr. Edges
91", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
16", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
16", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}