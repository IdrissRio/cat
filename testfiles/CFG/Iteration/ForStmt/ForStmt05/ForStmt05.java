public class ForStmt05 {
  {
    for (int b = 2, c = 1; b == 2; b++, c++) {
      System.out.println(2);
    }
  }
}