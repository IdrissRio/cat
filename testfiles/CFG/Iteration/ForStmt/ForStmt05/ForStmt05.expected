NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [IntegerLiteral[41]]
NodeName: IntegerLiteral[41] Successors: [VariableDeclarator[38]]
NodeName: VariableDeclarator[38] Successors: [IntegerLiteral[45]]
NodeName: IntegerLiteral[45] Successors: [VariableDeclarator[42]]
NodeName: VariableDeclarator[42] Successors: [VarAccess[48]]
NodeName: VarAccess[48] Successors: [IntegerLiteral[49]]
NodeName: IntegerLiteral[49] Successors: [EQExpr[47]]
NodeName: EQExpr[47] Successors: [TypeAccess[62], Exit[8]]
NodeName: TypeAccess[62] Successors: [VarAccess[63]]
NodeName: VarAccess[63] Successors: [IntegerLiteral[67]]
NodeName: IntegerLiteral[67] Successors: [MethodAccess[64]]
NodeName: MethodAccess[64] Successors: [PostIncExpr[52]]
NodeName: PostIncExpr[52] Successors: [PostIncExpr[55]]
NodeName: PostIncExpr[55] Successors: [VarAccess[48]]
NodeName: Exit[8] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [EQExpr[47]]
NodeName: EQExpr[47] Predecessor: [IntegerLiteral[49]]
NodeName: IntegerLiteral[49] Predecessor: [VarAccess[48]]
NodeName: VarAccess[48] Predecessor: [VariableDeclarator[42], PostIncExpr[55]]
NodeName: VariableDeclarator[42] Predecessor: [IntegerLiteral[45]]
NodeName: IntegerLiteral[45] Predecessor: [VariableDeclarator[38]]
NodeName: VariableDeclarator[38] Predecessor: [IntegerLiteral[41]]
NodeName: IntegerLiteral[41] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: PostIncExpr[55] Predecessor: [PostIncExpr[52]]
NodeName: PostIncExpr[52] Predecessor: [MethodAccess[64]]
NodeName: MethodAccess[64] Predecessor: [IntegerLiteral[67]]
NodeName: IntegerLiteral[67] Predecessor: [VarAccess[63]]
NodeName: VarAccess[63] Predecessor: [TypeAccess[62]]
NodeName: TypeAccess[62] Predecessor: [EQExpr[47]]
